src/PackageVersions/Installer.php

```diff
@@ -92,11 +92,17 @@ public static function dumpVersionsClass(Event $composerEvent)
         $io = $composerEvent->getIO();
         $io->write('<info>ocramius/package-versions:</info>  Generating version class...');
 
-        self::writeVersionClassToFile(
-            self::generateVersionsClass($versions),
-            $composer->getConfig(),
-            $composer->getPackage()
-        );
+        try {
+            self::writeVersionClassToFile(
+                self::generateVersionsClass($versions),
+                $composer->getConfig(),
+                $composer->getPackage()
+            );
+        } catch (NotInstalledException $exception) {
```

Comment:
Since you are using private-only API, throwing and catching in the same scope seems like a bad idea (it's basically a GOTO implemented via exceptions).

Can this be re-designed somehow?